pkg/activator/certificate/cache.go

```diff
@@ -114,3 +162,7 @@ func (cr *CertCache) handleCertificateUpdate(_, new interface{}) {
 func (cr *CertCache) GetCertificate(_ *tls.ClientHelloInfo) (*tls.Certificate, error) {
 	return cr.certificate, nil
 }
+
+func (cr *CertCache) GetClientCertificate(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
+	return cr.certificate, nil
+}
```

Comment:
`cr.certificate` has concurrent readers and writers and needs to be guarded

Reply:
cr.certificate is a pointer. 
When it gets updated, a different certificate is used and the actual pointer update is atomic. 
The certificate is created before cr.certificate is updated. 
The GetClientCertificate() is a reader that may return either the old certificate from before the change or the new one from after the change. Both are safe to use.

Reply:
> > actual pointer update is atomic.
> 
> This isn't true - what makes you think it is?

You mean like in 16 bit machines handling of 32 bin addresses?

If we want to cover such cases, we can add a lock here and make sure we return a healthy address. 
I will commit the change.

Reply:
> >  We will not have protection for cr.ClientTLSConf.RootCAs though.

> We have control which tls.Config is used in an http.Client when making outbound requests. If the CA changes we need to recreate these clients to use these new certs

nice, we can use this.
It does require that we move the lock such that it is available at the handler as well - I will move the lock to transport where the dialTLSContext is used.

Reply:
> You mean like in 16 bit machines handling of 32 bit addresses?

The go runtime makes no guarantee of these operations being atomic

Reply:
/retest